client/client.go

```diff
@@ -183,9 +183,9 @@ func (cli *Client) Close() error {
 
 // getAPIPath returns the versioned request path to call the api.
 // It appends the query parameters to the path if they are not empty.
-func (cli *Client) getAPIPath(p string, query url.Values) string {
+func (cli *Client) getAPIPath(p string, query url.Values, ignoreVersion bool) string {
```

Comment:
I think we still need to go through the rest of the function:

```go
	u := &url.URL{
		Path: apiPath,
	}
	if len(query) > 0 {
		u.RawQuery = query.Encode()
	}
	return u.String()
```

I don't like this solution either, but from `client.Ping` to `getAPIPath` it goes through rougly 4 functions. not sure how to handle that better. I could replace the `Context` part by a `bool` but not sure it would be enough ?